### DISCUSSION TITLE
.NET 8 / Oqtane 5.0

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#3400] [General]
For past major version upgrades I have always done the migration in the default Dev branch. The advantage to this approach is that it is much simpler to manage a single code base than multiple code bases and there is no extra effort required to migrate fixes between branches. However the disadvantage is that the Dev branch has the potential to be unstable for a period of time as the issues are resolved... requires upgraded tooling/environment (ie. VS2022, .NET 8 SDK, etc...)... and if you are used to building your own modules against the Dev branch it may impact your productivity (however in general it is always recommended to develop against official release packages and not the Dev branch so that you do not get affected).

So I am looking for some opinions for .NET 8 (remember the goal is to release Oqtane 5.0 in mid-November):

1. Perform the .NET 8 upgrade in the Dev branch 
2. Perform the .NET 8 upgrade in a separate ".NET 8" branch

Thoughts? ...

**Comment by leigh-pointer** [comment]
@sbwalker personally i would go for option 2. Also  I never build on a dev branch but I do test my compiled modules in the dev branch.

**Comment by sbwalker** [comment]
OK, there is a new "net8" branch available which contains the initial migration to .NET 8 (and Oqtane 5.0). It is a bit complicated to keep branches straight as you need to create a corresponding branch in your forked repo, clone your fork to a new location on disk, and then open the solution in the new location using Visual Studio 2022 Preview Edition, and point it at the "net8" branch.

EDIT: I am able to successfully install on LocalDB or SQLite

I can successfully create an additional site (ie. with Alias "localhost:44357/wasm") and specify WebAssembly as the hosting model

I can successfully run the MAUI solution (with the Oqtane Framework running in another VS2022 instance)

**Reply by thabaum** [reply]
Thank you. I was wondering what I needed to set for the package references.  This clears it up for me for next time I am testing a preview version.

**Reply by thabaum** [reply]
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/52467e57-fe55-4def-8517-b97f192c0afd)

firing off Oqtane.Server gives this error, should load a fresh install option screen.

**Reply by sbwalker** [reply]
@thabaum I am guessing you did not have a clean appsettings.json file? DefaultConnection needs to be "" in order to invoke the install wizard

**Reply by thabaum** [reply]
@sbwalker I submitted a PR to fix the issue.  Closed the other issue as no longer an issue.  Created a NET8 branch in my fork so I can provide any contributions as I did just now to test I am targeting  and using the correct branch.

I am going to play with the component render modes features now that everything seems to work and let you know when it breaks.  If I find any odd behaviors or broken I will test on a fresh install from this branch FYI.  I will also start developing modules and themes on this branch.

**Reply by sbwalker** [reply]
that was my fault... I did not follow the advice posted in the Readme:

If you want to submit pull requests it is recommended you ignore any local changes you have made to the appsettings.json file before you submit a pull request. To automate this activity, open a command prompt and navigate to the /Oqtane.Server/ folder and enter the command "git update-index --skip-worktree appsettings.json"

**Comment by sbwalker** [comment]
FYI... all development work will be in the new "net8" branch as we prepare for the Oqtane 5.0 release. The Dev branch will only be used if a true show-stopper issue comes up related to 4.x.

**Reply by thabaum** [reply]
This answers my questions, thanks I will target the other PR's to the net8 branch.

**Comment by hishamco** [comment]
Perform the .NET 8 upgrade in the Dev branch, but we can use multi-targeting instead of forcing everyone to use .NET 8.0

**Reply by sbwalker** [reply]
@hishamco this is a good suggestion. My only concern is that there are some major changes in .NET for Blazor which may require changes to the Oqtane project structure which I assume would not be backward compatible with .NET 7.

**Reply by hishamco** [reply]
I see, no problem with this case. We could continue with the latest version in both the `main` and `dev` branches, the `4.0.0` is suitable for those who want to use `net7.0`

In the future, we could use multi-targeting, we do the same thing in Orchard Core to support multiple frameworks

**Comment by hishamco** [comment]
Is the .NET 8.0 branch ready, we might need extensive tests to make sure everything is working as expected. Moreover, can we polish the admin UI if it's possible before the 5.0.0 release I planned for this in the past but I have an issue with the time. If it sound good I will work on this in a separated branch that could be merged if I finish before releasing 5.0.0

**Reply by thabaum** [reply]
@hishamco yes, visit the `net8` branch it has all the .NET 8 updates needed to test.  It includes all package dependencies and fixed any issues relating to .NET 8 upgrade of Oqtane solution projects.

I am doing all my testing and pull requests in .NET 8 currently.  We are not currently focused on "Framework" issues as much as we are reviewing .NET 8 issues so if it is not a "show stopper" don't expect much of a response.

I think for issues and pull requests we should label them starting with `.NET 8` so we know it is related 100% to .NET 8 and Oqtane Framework functionality.  I have posted a lot of issues and few pull requests but I don't expect them to get merged anytime soon as they are not .NET 8 breaking changes.

So far I have found none and .NET 8 has been backwards compatible with Oqtane Framework from testing I have done.  I have not done any "upgrades" or tests in this area and I know @sbwalker has been requesting these by anyone able.

I have been focused more on site setups, clean installs, site creation efforts along with any issues with general use over the past week.   I have also tested the module creator and theme creator.

I also tested using the Maui project with Window Desktop application and only tested with Windows, Visual Studio and MS SQL LocalDb scenarios FYI.

Azure, Linux, Mac, Chrome apps not tested on by me.  Not sure if anyone else is testing these environments.
